### DISCUSSION TITLE
How to implement function with named args?

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#8640] [Q&A] [answered]
I want to implement a function like this:

```sql
infer_schema(
    location=>'@mystage',
    file_format=>'my_parquet_format'
)
```

- How can I implement this kind of in the new expression framework?
- How to do async jobs here? Use `GlobalIoRuntime.block_on`?

cc @andylokandy @sundy-li for comments

related to https://github.com/datafuselabs/databend/issues/6345

**Comment by andylokandy** [comment, marked as answer]
Are there many of them?

**Reply by Xuanwo** [reply]
No, only one so far.

**Reply by Xuanwo** [reply]
By using named args, we can extend the inputs without breaking users. For example:

```sql
infer_schema(
    location=>'@mystage',
    connection = ( endpoint_url='https://example.com' )
    file_format=>'my_parquet_format'
)
```

Maybe it's not a good idea to be designed as a function?

We also want to support create table via `infer_schema`, for example:

```sql
create table mytable
  using template (
    select array_agg(object_construct(*))
    within group (order by order_id)
      from table(
        infer_schema(
          location=>'@mystage',
          file_format=>'my_parquet_format'
        )
      ));
```

If it possible if we implement `infer_scehma` as a SQL statement?

**Reply by Xuanwo** [reply]
Current design: https://github.com/datafuselabs/databend/pull/8645

**Reply by andylokandy** [reply]
seems like you can add a new expr element to ast, and then desugar it to function call in the binder.
